taurus/demo/cake.py

```diff
@@ -126,27 +136,38 @@ def make_cake():
 
     # and abstract ingredients
     for run in cake.process.ingredients:
-        run.spec = IngredientSpec(material=run.material.spec, process=cake.process.spec)
+        run.spec = IngredientSpec(name='{} input'.format(run.material.spec.name),
+                                  material=run.material.spec, process=cake.process.spec)
 
     for run in baked.process.ingredients:
-        run.spec = IngredientSpec(material=run.material.spec, process=baked.process.spec)
+        run.spec = IngredientSpec(name='{} input'.format(run.material.spec.name),
+                                  material=run.material.spec, process=baked.process.spec)
 
     for run in batter.process.ingredients:
-        run.spec = IngredientSpec(material=run.material.spec, process=batter.process.spec)
+        run.spec = IngredientSpec(name='{} input'.format(run.material.spec.name),
+                                  material=run.material.spec, process=batter.process.spec)
 
     for run in wetmix.process.ingredients:
-        run.spec = IngredientSpec(material=run.material.spec, process=wetmix.process.spec)
+        run.spec = IngredientSpec(name='{} input'.format(run.material.spec.name),
+                                  material=run.material.spec, process=wetmix.process.spec)
 
     for run in drymix.process.ingredients:
-        run.spec = IngredientSpec(material=run.material.spec, process=drymix.process.spec)
+        run.spec = IngredientSpec(name='{} input'.format(run.material.spec.name),
+                                  material=run.material.spec, process=drymix.process.spec)
 
     set_uuids(cake)
 
-    IngredientSpec(material=LinkByUID.from_entity(butter.spec), process=frosting.spec.process)
-    IngredientSpec(material=chocolate.spec, process=frosting.spec.process)
-    IngredientSpec(material=powder_sugar.spec, process=frosting.spec.process)
-    IngredientSpec(material=LinkByUID.from_entity(vanilla.spec), process=frosting.spec.process)
-    IngredientSpec(material=LinkByUID.from_entity(milk.spec), process=frosting.spec.process)
+    print(butter)
+
+    for run in frosting.process.ingredients:
```

Comment:
Suggest changing `run` to `ingredient_run`, for readability

Reply:
Changing to ing_run as compromise, given flake8 line length constraint.